`__STDC_WANT_LIB_EXT1__` defined after standard library headers are included, causes build failure

### Generated by Generative AI

Assisted by generative AI for debugging and translation.

### Operating System:

Ubuntu 22.04 x86_64

### ROS version or commit hash:

humble and others

### 'ros2 doctor --report' output

ROS2 environment not directly activated.

### Steps to reproduce issue

> Set up the ESP-IDF release/v6.0 environment as described in the `micro_ros_espidf_component` README, then build the `int32_publisher` example from the `humble` branch.

I'm trying to adapt [micro_ros_espidf_component](https://github.com/micro-ROS/micro_ros_espidf_component) to ESP-IDF 6.0, which uses `picolibc` with `Annex K` support by default. During the build process, a lot of errors like this appear:

```
In file included from ~/esp/esp-idf/components/esp_libc/platform_include/assert.h:12,
                 from ~/workshop/micro-ros/micro_ros_espidf_component/micro_ros_src/src/rcutils/include/rcutils/error_handling.h:28,
                 from ~/workshop/micro-ros/micro_ros_espidf_component/micro_ros_src/src/rcutils/src/hash_map.c:24:
~/.espressif/tools/riscv32-esp-elf/esp-15.2.0_20251204/riscv32-esp-elf/picolibc/include/stdlib.h:415:9: error: unknown type name '__errno_t'; did you mean '__ino_t'?
  415 | typedef __errno_t errno_t;
      |         ^~~~~~~~~
      |         __ino_t
```

This is because `error_handling.h` contains the `Annex K` macro `__STDC_WANT_LIB_EXT1__`, which needs to be defined before any standard library headers are included. 

https://github.com/ros2/rcutils/blob/cf916d5aa9bfe0655e499d269a88e6053a3eea42/include/rcutils/error_handling.h#L25-L27

But many other files include standard library headers first and then `error_handling.h`, so some `Annex K` types end up undefined.

https://github.com/ros2/rcutils/blob/0afe4f1b2a192a7e1bdf133ea0d616a7f6203d61/src/hash_map.c#L20-L24

Relevant content in `stdlib.h` (error location):

```C
#if __STDC_WANT_LIB_EXT1__ == 1
#include <sys/_types.h>

#ifndef _ERRNO_T_DEFINED
typedef __errno_t errno_t;
#define _ERRNO_T_DEFINED
#endif
```

### Expected behavior

Build succeeded

### Actual behavior

Build failed

### Additional information

`micro_ros_espidf_component` is actually using a [fork](https://github.com/micro-ROS/rcutils) of this repository, but the issue is the same.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,6 +81,7 @@ target_include_directories(${PROJECT_NAME} PUBLIC
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
 target_compile_definitions(${PROJECT_NAME} PRIVATE "RCUTILS_BUILDING_DLL")
+target_compile_definitions(${PROJECT_NAME} PUBLIC __STDC_WANT_LIB_EXT1__=1)
 
 if(BUILD_TESTING AND NOT RCUTILS_DISABLE_FAULT_INJECTION)
   target_compile_definitions(${PROJECT_NAME} PUBLIC RCUTILS_ENABLE_FAULT_INJECTION)

--- a/src/error_handling_helpers.h
+++ b/src/error_handling_helpers.h
@@ -31,9 +31,6 @@
 #define RCUTILS_WARN_ON_TRUNCATION 1
 #endif
 
-#ifndef __STDC_WANT_LIB_EXT1__
-#define __STDC_WANT_LIB_EXT1__ 1  // indicate we would like memmove_s if available
-#endif
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>

